### ISSUE TITLE
Sparql data stored by database-ms has subject, predicate and object as URL's

### ISSUE THREAD

**Issue by prasanthhs** [dice-group/sask#60]
Are there  any extractors which index labels against URL's in the sparql db from the current code? I have tried open IE MS and it returns URL's for all the three fields of triples. I am not sure about other extractors.

We had a discussion with Ricardo about this and he mentioned that Auto Index will index label vs URL and not URL vs URL.

Some sample FOX answers(shared by Ricardo):

Query : Barack Obama is  born in Hawaii 
[FOX_output_for__Barack_Obama_is_born_in_Hawaii__.txt](https://github.com/dice-group/sask/files/2144504/FOX_output_for__Barack_Obama_is_born_in_Hawaii__.txt)

They store only
foxr:1530176121568  a  oa:Annotation , rdf:Statement , foxo:Relation ;
        rdf:object     dbr:Hawaii ;
        rdf:predicate  foxo:stanford_livein ;
        rdf:subject    dbr:Barack_Obama ;

and should create  this   dbr:Barack_Obama rdfs:label  dbr:Hawaii .

Finally, Label vs URL has to be indexed in Auto Index and not URL vs URL.

Any suggestions on how to proceed further?

**Comment by KHaack** [comment]
Hmm one possible solution is to use jena to parse the data. The  second is, that we parse every output of any extractor before we insert the data into the database or select an other output format if possible (and its sometimes not possible)

**Comment by RicardoUsbeck** [comment]
Parsing it into a Jena Model plus running a query. Otherwise too much data ends up as garbage in the score. That should also concern @hjshah142

**Comment by sepidetari** [comment]
ok so you need something like this?
`<br/>
```subject (as uri) -----predicate (as label)-----> object (as uri)```

**Comment by RicardoUsbeck** [comment]
From "Barack Obama is born in Hawaii" the following should be sent to the triple store:
```
dbr:Barack_Obama foxo:stanford_livein dbr:Hawaii.
dbr:Barack_Obama rdfs:label "Barack Obama" .
dbr:Hawaii rdfs:label "Hawaii" .
foxo:stanford_livein rdfs:label "born in".
```

**Comment by sepidetari** [comment]
@RicardoUsbeck Does It mean that we just have to store turtle format  in DB?<br/>
We parsed the ttl into Jena model and stored it in DB, the following pictures are the result of it. Is this fine?
![db 1](https://user-images.githubusercontent.com/24546007/42048882-bc10c32e-7b04-11e8-9350-562b45722211.jpg)
![db 2](https://user-images.githubusercontent.com/24546007/42048885-be7bd93c-7b04-11e8-8e61-ebb1056dcf2f.jpg)

**Comment by Suganya31** [comment]
@RicardoUsbeck @prasanthhs If this looks fine, we can try making the open IE and Sorookin to give output in the form of TTL.

**Comment by RicardoUsbeck** [comment]
Would be great. Is that also possible for FOX so that harsh has three annotators for the ensemble learning?

**Comment by RicardoUsbeck** [comment]
No it does not since there is no triple that can bind to ?key1 a owl:Thing . 

```
SELECT DISTINCT ?key1 ?key2 
WHERE{
?key1 rdfs:label ?key2 .
FILTER(regex(str(?key1), "/ontology/[a-z]" ))}
```
This one should work and return labels for propertie for example

**Comment by prasanthhs** [comment]
@RicardoUsbeck yes i know. But the problem is that that query is a generic default query which runs for all remote end points.. Should i make the query you've given the default instead for only local sparql end points? What about queries for classes and properties which are also executed against this local database?

Plus there's the case of missing Prefixes. In the extractor's data there are prefixes which are not included by default with auto Index. Isn't it better if the query is passed instead with necessary prefixes ( opposite to what we discussed) ?

**Comment by Suganya31** [comment]
@RicardoUsbeck  I am not sure about FOX but I can try Sorookin and Open IE. @KHaack  Is it possible for FOX also?

**Comment by RicardoUsbeck** [comment]
@prasanthhs the problem is, that the SASK knowledge graph is not complete but yes, you can also configure it so the query is passed to the autoindex

**Comment by prasanthhs** [comment]
@RicardoUsbeck Yes I understand..Then, it is probably better for database-ms to pass the new query posted here instead of configuring auto index since this is a sask related work in progress. Once the complete implementation is done, the custom queries can be removed.

**Comment by Suganya31** [comment]
Fox stores the proper TTL in the database now.

**Comment by prasanthhs** [comment]
Snapshot of Elastic Search Repository for FOX + Auto Index integration. 

Query passed: SELECT DISTINCT ?key1 ?key2 WHERE{?key1 rdfs:label ?key2 . 
![screen shot 2018-07-03 at 13 03 56](https://user-images.githubusercontent.com/33149420/42216315-be4a94ec-7ec1-11e8-9550-1b78e4da6779.png)

**Comment by RicardoUsbeck** [comment]
If we could get the label of the property in the future it would be even better, but currently FOX does not return them. Maybe @Suganya31 can open an issue for that
